### PR TITLE
[sc-10746] Improve LLM user key management

### PR DESCRIPTION
**Pull request by operramon**

### CODE REVIEW

File: libs/common/src/lib/ai-models/answer-generation/answer-generation.component.ts [hunk 1]
```diff
@@ -58,8 +59,13 @@ export class AnswerGenerationComponent extends LearningConfigurationDirective {
     }),
   });
   currentGenerativeModel?: LearningConfigurationOption;
-  hasOwnKey = false;
+  userKeyToggle = new FormControl<boolean>(false);
+  required = this.translate.stream('kb.ai-models.common.required');
```
Comment (mpellerin42) [inline libs/common/src/lib/ai-models/answer-generation/answer-generation.component.ts:63]: why using a stream here? with `instant` you would not need `| async` in the template

File: libs/common/src/lib/ai-models/answer-generation/answer-generation.component.html [hunk 2]
```diff
@@ -54,13 +63,19 @@
...
+                        ">
                         {{ field.value.title }}
                       </pa-textarea>
                     } @else {
                       <pa-input
                         [formControlName]="field.key"
-                        [help]="field.value.info">
+                        [help]="
+                          (userKeys.required.includes(field.key) ? (required | async) + ' ' : '') +
+                          (field.value.info || '')
```
Comment (mpellerin42) [inline libs/common/src/lib/ai-models/answer-generation/answer-generation.component.html:77]: if there is no `info`, we will end up with an empty `()`

Comment (mpellerin42) [reply libs/common/src/lib/ai-models/answer-generation/answer-generation.component.html]: I just realised my mistake, those `()` are for the expression, not a string. So forgot about this comment

**Review by mpellerin42 — APPROVED** [review]

File: libs/common/src/lib/ai-models/answer-generation/answer-generation.component.ts (hunk 1 above)
Comment (operramon) [reply libs/common/src/lib/ai-models/answer-generation/answer-generation.component.ts]: Fixed

File: libs/common/src/lib/ai-models/answer-generation/answer-generation.component.html (hunk 2 above)
Comment (operramon) [reply libs/common/src/lib/ai-models/answer-generation/answer-generation.component.html]: I just tested it. It seems fine, the parentheses are not rendered

**Pull request merged**